lib/internal/crypto/crypto.js

```diff
@@ -0,0 +1,527 @@
+'use strict';
+
+const binding = process.binding('crypto');
+const constants = process.binding('constants').crypto;
+const cryptoUtil = require('internal/crypto/util');
+const LazyTransform = require('internal/streams/lazy_transform');
+const internalUtil = require('internal/util');
+const StringDecoder = require('string_decoder').StringDecoder;
+const Writable = require('stream').Writable;
+const Buffer = require('buffer').Buffer;
+const assert = require('assert');
+const util = require('util');
+const toBuf = cryptoUtil.toBuf;
+const inspect = internalUtil.customInspectSymbol;
+
+const kAlgorithm = Symbol('algorithm');
+
+const DH_GENERATOR = 2;
+
+class HashBase extends LazyTransform {
```

Comment:
Moved and refactored in the ES6 class. There are no *functional* changes with the exception of a small bug fix.